src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/ContentPickerFieldIndexHandler.cs

```diff
@@ -10,11 +10,20 @@ public override Task BuildIndexAsync(ContentPickerField field, BuildFieldIndexCo
         {
             var options = DocumentIndexOptions.Store;
 
-            foreach (var contentItemId in field.ContentItemIds)
+            if (field.ContentItemIds.Length > 0)
             {
+                foreach (var contentItemId in field.ContentItemIds)
+                {
+                    foreach (var key in context.Keys)
+                    {
+                        context.DocumentIndex.Set(key, contentItemId, options);
+                    }
+                }
+            }
+            else {
                 foreach (var key in context.Keys)
                 {
-                    context.DocumentIndex.Set(key, contentItemId, options);
+                    context.DocumentIndex.Set(key, "NULL", options);
```

Comment:
I think the field should not be set if there is a null value, for any type of field. Would that work?

Reply:
I understand what you mean, I need to try different things here. Somehow I required a NULL value to be indexed to be able to find content items that had a NULL value set. I need to re-run those Queries and evaluate.

Reply:
And with that we could use queries like this:

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-exists-query.html